src/transformer/utils/get_multichoice_definition.php

```diff
@@ -18,36 +18,42 @@
 defined('MOODLE_INTERNAL') || die();
 
 function get_multichoice_definition(array $config, \stdClass $questionattempt, \stdClass $question, $lang) {
+    // For some reason, newlines and &nbsp is being added to strings,
+    // in order to remove new lines we have to ensure nbsp is also removed.
+    $replacestrings = array("\n", "&nbsp");
+    // Reducing number of times string replace is called, by moving duplicate code up here.
+    $questiontext = str_replace($replacestrings, "", strip_tags($question->questiontext));
```

Comment:
Can we please extract this logic into a util function called "cleanHtmlText" or something like that? Thinking we may need it elsewhere and it keeps the transformers a little cleaner with less code duplication.

Reply:
Yes, doing that now.